source/d3d11/d3d11_device_context.cpp

```diff
@@ -16,23 +16,254 @@ void D3D11DeviceContext::log_drawcall(UINT vertices)
+	{
+		if (pDepthStencilView != _active_depthstencil)
+		{
+			_active_depthstencil = pDepthStencilView;
+			_draw_call_tracker.track_depthstencil(_active_depthstencil.get());
+		}
+	}
+}
+
+void D3D11DeviceContext::set_active_OM_depthstencil(ID3D11DepthStencilView* pDepthStencilView)
+{
+	if (pDepthStencilView != nullptr)
+	{
+		com_ptr<ID3D11Texture2D> texture;
+		com_ptr<ID3D11Resource> resource;
+		D3D11_TEXTURE2D_DESC texture_desc;
+
+		pDepthStencilView->GetResource(&resource);
+		if (!FAILED(resource->QueryInterface(&texture)))
+		{
+			texture->GetDesc(&texture_desc);
+		}
+		else
+		{
+			return;
+		}
+
+		if (!check_depthstencil(texture, texture_desc))
+		{
+			return;
+		}
+
+		if (_OM_iter >= _clear_DSV_iter)
+		{
+			set_active_depthstencil(pDepthStencilView);
+			// copy the depth stencil texture
+			_depth_texture = texture.get();
+
+			if (_depth_texture != nullptr)
+			{
+				_draw_call_tracker.set_depth_texture(_depth_texture.get());
+			}
+		}
+	}
+}
+
+void D3D11DeviceContext::set_active_cleared_depthstencil(ID3D11DepthStencilView* pDepthStencilView)
+{
+	if (pDepthStencilView != nullptr)
 	{
-		_active_depthstencil = pDepthStencilView;
+		com_ptr<ID3D11Texture2D> texture;
+		com_ptr<ID3D11Resource> resource;
+		D3D11_TEXTURE2D_DESC texture_desc;
 
-		if (pDepthStencilView != nullptr)
+		pDepthStencilView->GetResource(&resource);
+		if (!FAILED(resource->QueryInterface(&texture)))
 		{
-			_draw_call_tracker.track_depthstencil(pDepthStencilView);
+			texture->GetDesc(&texture_desc);
+		}
+		else
+		{
+			return;
+		}
+
+		if (!check_depthstencil(texture, texture_desc))
+		{
+			return;
+		}
+
+		reshade::d3d11::draw_call_tracker::depthstencil_counter_info counters = _draw_call_tracker.get_counters();
+
+		if (reshade::runtime::depth_buffer_clearing_number == 0)
+		{
+			set_active_depthstencil(pDepthStencilView);
+			// copy the depth stencil texture
+			_depth_texture = copy_depth_texture(texture_desc, texture.get());
+
+			if (_depth_texture != nullptr)
+			{
+				_draw_call_tracker.set_depth_texture(_depth_texture.get());
+			}
+		}
+		else if(reshade::runtime::depth_buffer_clearing_number == _clear_DSV_iter)
+		{
+			if (counters.vertices > _VERTICES_TRESHOLD && counters.vertices > _best_vertices)
+			{
+				set_active_depthstencil(pDepthStencilView);
+				// copy the depth stencil texture
+				_depth_texture = copy_depth_texture(texture_desc, texture.get());
+
+				if (_depth_texture != nullptr)
+				{
+					_draw_call_tracker.set_depth_texture(_depth_texture.get());
+				}
+
+				_best_vertices = counters.vertices;
+				_best_drawcalls = counters.vertices;
+				_clear_DSV_iter++;
+			}
 		}
 	}
 }
 
 void D3D11DeviceContext::clear_drawcall_stats()
 {
-	_draw_call_tracker.reset();
+	_clear_DSV_iter = 1;
+	_OM_iter = 0;
+	_best_vertices = 0;
+	_best_drawcalls = 0;
+	_draw_call_tracker.reset(true);
 	_active_depthstencil.reset();
+	_depth_texture.reset();
+}
+
+void D3D11DeviceContext::screen_dimensions()
```

Comment:
I don't see the reason behind this function and the `_screen_width` + `_screen_height` class fields. They are used in a single function only, so may as well store them on the stack instead.

Reply:
changed